pinot-core/src/main/java/org/apache/pinot/core/data/function/DateTimeFunctions.java

```diff
@@ -27,6 +32,8 @@
  */
 public class DateTimeFunctions {
 
+  private Map<String, DateTimeFormatter> _dateTimeFormatterMap = new HashMap<>();
```

Comment:
While this is working great for dateTimeFormat functions,  I wonder if we need to instead have multiple constructors (or init methods), and create DateTimeFunction object for every function we encounter in the schema.
For example, say we would want to add `round(millis, 15:MINUTES)`. The logic for this is (millis / (15*60*1000)) * (15*60*1000). We don't want to interpret "15:MINUTES" on every row. Instead, we can create a DateTimeFunction that saves "15:MINUTES" as a member variable, and then invoke `round()` on this object.
We can have init methods like
initBucket(int bucket), initRounding(int rounding), initSDF(String pattern), etc.

Reply:
As per offline discussion, we will focus only on toDateTime for now. This comment can be taken up as a separate item in the future